library/std/src/sys/unix/locks/pthread_condvar.rs

```diff
@@ -110,25 +104,14 @@ impl Condvar {
         target_os = "horizon"
     )))]
     pub unsafe fn wait_timeout(&self, mutex: &Mutex, dur: Duration) -> bool {
-        use crate::mem;
-
-        let mut now: libc::timespec = mem::zeroed();
-        let r = libc::clock_gettime(libc::CLOCK_MONOTONIC, &mut now);
-        assert_eq!(r, 0);
-
-        // Nanosecond calculations can't overflow because both values are below 1e9.
-        let nsec = dur.subsec_nanos() + now.tv_nsec as u32;
-
-        let sec = saturating_cast_to_time_t(dur.as_secs())
-            .checked_add((nsec / 1_000_000_000) as libc::time_t)
-            .and_then(|s| s.checked_add(now.tv_sec));
-        let nsec = nsec % 1_000_000_000;
-
-        let timeout =
-            sec.map(|s| libc::timespec { tv_sec: s, tv_nsec: nsec as _ }).unwrap_or(TIMESPEC_MAX);
+        use crate::sys::time::Timespec;
 
+        let timeout = Timespec::now(libc::CLOCK_MONOTONIC)
+            .checked_add_duration(&dur)
+            .and_then(|t| t.to_timespec())
+            .unwrap_or(TIMESPEC_MAX);
         let r = libc::pthread_cond_timedwait(self.inner.get(), pthread_mutex::raw(mutex), &timeout);
-        assert!(r == libc::ETIMEDOUT || r == 0);
+        debug_assert!(r == libc::ETIMEDOUT || r == 0);
```

Comment:
Why the change from assert! here?

Reply:
Because all other errors are only emitted by some implementations, but the incorrect usage they are indicating may also cause UB. Hence, this assertion is only really useful as a simple smoke test in debug builds. This also matches the assert in `wait`.

Reply:
I personally feel like we should keep this as an assert, but I'm not sure what the current consensus there is - I know there was some discussion recently about this....

https://github.com/rust-lang/rust/pull/99624 seems like it generally moves us in the direction of preferring assert, and I would want that myself, so I think we should stick with asserts.

Reply:
Should I update the other assertions too?

Reply:
I've now kept all assertions as they were.